lib/cpp/src/thrift/transport/TSocketPool.cpp

```diff
@@ -188,7 +189,13 @@ void TSocketPool::open() {
   }
 
   if (randomize_ && numServers > 1) {
-    random_shuffle(servers_.begin(), servers_.end());
+#if __cplusplus >= 201500L // c++17
```

Comment:
201703L is the correct constant for C++17; this one may include some pre-C++17 standard compilers (6.1 for example) and we don't test with a wide variety of them.